### PR TITLE
Re 6470/enable special case of config files in the app data folder

### PR DESCRIPTION
**Pull request by McdonaldSeanp**
This is direct cut over of parts of https://github.com/puppetlabs/puppet_for_the_win/blob/master/wix/puppet.wxs as specific static files pertaining to the special case for config files in CommonAppDataFolder

### CODE REVIEW

**Comment by jcoconnor** [comment]
@McdonaldSeanp This looks sound but I need to build it and test it tomorrow to confirm.

File: resources/windows/wix/appdatafiles.wxs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          <Directory Id="EnvironmentsDir" Name="environments">
+            <Directory Id="ProductionDir" Name="production">
+              <Directory Id="ManifestsDir" Name="manifests"/>
+              <Directory Id="ModulesDir" Name="modules"/>
+            </Directory>
+          </Directory>
+        </Directory>
+      </Directory>
+    </DirectoryRef>
+    <ComponentGroup Id="AppDataComponentGroup" >
```
Comment (jcoconnor) [inline resources/windows/wix/appdatafiles.wxs]: Can you indent all the component lines as well - really anything with 3 or more items should be indented for clarity.

File: resources/windows/wix/appdatafiles.wxs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            </Directory>
+            <Directory Id="pxpAgentDataFolderEtc" Name="etc">
+              <Directory Id="pxpAgentDataFolderModules" Name="modules"/>
+            </Directory>
+          </Directory>
+          <Directory Id="PuppetAppData" Name="puppet">
+            <Directory Id="PuppetConfDir" Name="etc"/>
+            <Directory Id="PuppetVarDir" Name="var"/>
+          </Directory>
+        </Directory>
```
Comment (jcoconnor) [inline resources/windows/wix/appdatafiles.wxs]: @McdonaldSeanp This line belongs just before line 45.
If you check the install you will see that  ```code```, ```facter``` and ```mcollective``` are being created under ```ProgramData``` and not ```PuppetLabs```

**Comment by jcoconnor** [comment]
@McdonaldSeanp Have tested this and it appears to install ok, with the one exception that 3 of the directories are placed under ProgramData instead of PuppetLabs - have marked the line where I think you need to move a ```</Directory>``` statement.

The ProgramData directories and configuration files are left in place after the uninstall which is expected.

Fixing the Component indentation would also make this a lot more readable especially under githubs insanely narrow review windows.

**Comment by jcoconnor** [comment]
Sean - it would be good to add some more documentation at the top of the file as to why these files are being given the special treatment.

**Comment by jcoconnor** [comment]
Have just tested in build and install - the ProgramData issue has been fixed.

:+1:

**Comment by ferventcoder** [comment]
LGTM :+1:

**Pull request merged**